mantidproject/mantid
# Saving the muon analysis 2 GUI

This PR is to add the ability to save a python GUI (muon analysis 2) as part of project saving. I will add some documentation on how to save a GUI in #23836 and it does not work with project recovery #23837

**To test:**
Before compiling you will need to enable muon analysis 2 by adding:
`Muon/Muon_Analysis_2.py` to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L23

Once it has built:
1. Open Mantid and load any workspace
2. Open Muon Analysis 2
3. Edit the interface state (change group names etc.)
4. Save the project and make sure to include both the ws and GUI
5. Close Mantid
6. Open Mantid and load in the project we just saved
7. Muon Analysis 2 should open with the values you had previously entered. 

If you feeling adventures you can go into the project file and change some of the values. 



Fixes #19150. 




---
This does not require release notes
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
